subgraphs/_reference_/src/common/utils/logger.ts

```diff
@@ -1,26 +1,50 @@
 import { ethereum, log } from "@graphprotocol/graph-ts";
 
+/**
+ * Logger class is used to log messages to be picked up by our logging system.
+ *
+ * @param event The event that triggered the log message
+ * @param funcName The name of the function that triggered the log message
+ * @param msg The message to log (add `{}` to the message to add additional arguments)
+ * @param args Any additional arguments to add to the msg (same format as graph-ts log functions)
+ */
 export class Logger {
   event: ethereum.Event;
+  fnName: string | null;
 
-  constructor(event: ethereum.Event) {
+  constructor(event: ethereum.Event, fnName: string | null = null) {
```

Comment:
In which circumstances we'd want to allow a null function name?

Can we force the user to always set a fnName?

Reply:
never